File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/data/repository/CidadeRepositoryTask.java
package com.example.agnciadeturismo.data.repository;

import android.util.Log;

import com.example.agnciadeturismo.data.api.RetrofitTask;
import com.example.agnciadeturismo.model.CidadeDto;

import java.io.IOException;
import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Response;

public class CidadeRepositoryTask {

    private static final String TAG = "CidadeRepositoryTask";
    private static ArrayList<String> listCidade = new ArrayList<>();
    private static Integer cdOrigem;
    private static Integer cdDestino;

    public static ArrayList<String> getAllCidades(){

        Call<ArrayList<CidadeDto>> api = RetrofitTask.getRetrofit().getAllCidades();
        try {
            Response<ArrayList<CidadeDto>> response = api.execute();
            if(response.isSuccessful()){
                for(CidadeDto cidade : response.body()){
                    Log.d(TAG, "Quantidade");
                    listCidade.add(cidade.getCidade());
                    Log.d(TAG, "Quantidade");
                }
                Log.d(TAG, "Quantidade: "+listCidade.size());
            }else{
                Log.d(TAG, "ERRO: "+response.code());
            }
        } catch (IOException e) {
            Log.d(TAG, "ERRO: "+e.getMessage());
        }

        return listCidade;
    }

    public static Integer getOrigemCidade(String origem){
        Call<ArrayList<CidadeDto>> api = RetrofitTask.getRetrofit().getNomeCidade(origem);
        try {
            Response<ArrayList<CidadeDto>> response = api.execute();
            if(response.isSuccessful()){
                ArrayList<CidadeDto> listCidade = response.body();
                if(listCidade.size() != 0){
                    cdOrigem = listCidade.get(0).getCd();
                }else{
                    cdOrigem = null;
                    Log.d(TAG, "Não encontrado a cidade origem");
                }

            }else{
                Log.d(TAG, "ERRO: "+response.code());
            }
        } catch (IOException e) {
            Log.d(TAG, "ERRO: "+e.getMessage());
        }

        return cdOrigem;
    }

    public static Integer getDestinoCidade(String nome){

        Call<ArrayList<CidadeDto>> api = RetrofitTask.getRetrofit().getNomeCidade(nome);
        try {
            Response<ArrayList<CidadeDto>> response = api.execute();
            if(response.isSuccessful()){
                ArrayList<CidadeDto> listCidade = response.body();
                if(listCidade.size() != 0){
                    cdDestino = listCidade.get(0).getCd();
                }else{
                    cdDestino = null;
                    Log.d(TAG, "Não encontrado a cidade destino");
                }

            }else{
                Log.d(TAG, "ERRO: "+response.code());
            }
        } catch (IOException e) {
            Log.d(TAG, "ERRO: "+e.getMessage());
        }

        return cdDestino;
    }
}

File: AgenciaTurismo/app/src/main/java/com/example/agnciadeturismo/data/repository/CidadeRepositoryTask.kt
package com.example.agnciadeturismo.data.repository

import android.util.Log
import com.example.agnciadeturismo.data.api.RetrofitTask
import java.util.*

class CidadeRepositoryTask {

    private val TAG = "CidadeRepositoryTask"
    private val listCidade = ArrayList<String>()
    private var codigoCidade: Int? = null
    private var cidade: String? = null

    fun getAllCidades(): ArrayList<String> {
        val response = RetrofitTask.getRetrofit().allCidades.execute()
        if(response.isSuccessful){
            for (cidade in response.body()!!) {
                listCidade.add(cidade.cidade)
            }
            Log.d(TAG, "Quantidade: " + listCidade.size)
        }else{
            Log.d(TAG, response.errorBody().toString())
        }

        return listCidade
    }

    fun getCodigoCidade(cidade: String?): Int? {
        val response = RetrofitTask.getRetrofit().getNomeCidade(cidade).execute()
        if (response.isSuccessful) {
            val listCidade = response.body()!!
            if (listCidade.size != 0) {
                codigoCidade = listCidade[0].cd
            } else {
                codigoCidade = null
                Log.d(TAG, "Não encontrado a cidade")
            }
        } else {
            Log.d(TAG, "ERRO: " + response.code())
        }

        return codigoCidade
    }

    fun getNomeCidade(codigo: Int): String? {
        val response = RetrofitTask.getRetrofit().getCodigoCidade(codigo).execute()
        if (response.isSuccessful) {
            val listCidade = response.body()!!
            if (listCidade.size != 0) {
                cidade = listCidade[0].cidade
            } else {
                cidade = null
                Log.d(TAG, "Não encontrado a cidade")
            }
        } else {
            Log.d(TAG, "ERRO: " + response.code())
        }

        return cidade
    }

}